network/limitlistener/rejectingLimitListener_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2013 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package limitlistener_test
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/algorand/go-algorand/network/limitlistener"
+	"github.com/algorand/go-algorand/test/partitiontest"
+)
+
+const defaultMaxOpenFiles = 256
+const timeout = 5 * time.Second
+
+func TestRejectingLimitListener(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	const max = 5
+	attempts := (maxOpenFiles() - max) / 2
+	if attempts > 256 { // maximum length of accept queue is 128 by default
+		attempts = 256
+	}
+
+	l, err := net.Listen("tcp", "127.0.0.1:0")
```

Comment:
This looks like it should work, but you could probably implement it with some sort of `nonClosingListener`, similar to your `errorListener` to make it a little clearer and avoid the `time.Sleep`.

Reply:
Probably. I copied the tests too and decided not to rewrite them.